### ISSUE TITLE
New official releases (last is 4 yrs old!)

### ISSUE THREAD

**Issue by GWRon** [bmx-ng/bmx-ng#190]
Heya,

as some people at discord seem to prefer the downloads which look "official" - so from blitzmax.org, they will use these (v0.129) and thus also report issues fixed up to 4 years ago.

I think we should use one of the latest weekly build revisions to make a new official release. Current issues (Mac M1 building etc) are issues in the old "official release" too, so changing it to a recent one will not break too much.

If certain issues need to be resolved first, I would suggest that @woollybah simply adds a todo-list to here which he updates with commits fixing these todo points.

**Comment by MidimasterSoft** [comment]
I think weekly RELEASE is too often. And a OFFICIAL RELEASE should not processed by automatic, but a by human. Somebody needs to decide if the current state is a stable one or had a lot of changes and needs first be verified by the experienced users.

So I prefer a periodical Release Update within a period of 2-4 month, motivated by the experience of users, that this release starts perfect and has no "construction sites".

**Comment by GWRon** [comment]
I did not write anything about using a weekly release each time as an official release.

I wrote "we should use _ONE_ of the latest weekly build revisions". So yes, it should be one at least some people are using "productively" already. And it should contain critical bugfixes.

bcc - not changed for 4 months, so all the latest weekly builds use it
bmk - not changed for 10 months, so all the latest weekly builds use it

brl.mod - no fixes, just additions of functionality (old users won't use them, so won't break existing code)
text.mod - same
pub.mod - no changes
archive.mod - no changes in 2024
sdl.mod - lib updated (a while ago) and since then only fixes (so "can only get better the more recent it is")


I do not see any reason not to use the latest (as of now :D) weekly build revision / date / ... including the modules to build an "official release".

"normally" Brucey could simply select/mark a weekly build to become a "stable build" and the website _could_ automatically use the latest stable builds. So this must not be automatically, it can happen "on click". Of course it needs preparations.
All more sophisticated solutions involve "hashes" of the module repositories so a build script knows which revision to use for a build. Then "releases" are more a kind of txt/ini file containing all the module repos + revisions to use.
Yet ...I think for now simply creating a new and more recent release of all the major OSes should be preferred over "overthinking" all this release cycles ideas.

**Comment by meemoeuk2** [comment]
I recently downloaded BlitzMax from blitzmax.org
https://github.com/bmx-ng/bmx-ng/releases/download/v0.136.3.51.win32.mingw/BlitzMax_win32_mingw_0.136.3.51.7z
which suggests the version is v0.136.3.51
but in the versions.txt that comes with the download it says the version is 1.50
If I use the 'about' tab in the downloaded MaxIDE it says the versions are
MaxIDE version is 1.56 (ng)
bcc[ng] version is 0.136
bmk version is 3.51

I had to search around for clues on the github, not sure, but I think these are fairly recent versions of blitzmax ng files.

**Comment by GWRon** [comment]
you downloaded an old version (releases -- https://github.com/bmx-ng/bmx-ng/releases )

the version numbers in the download represent multiple things:
- MaxIDE has its own version
- bcc version is the first part of the download version (0.136)
- bmk version the second (3.51)
- bcc + bmk = "v0.136.3.51"

So these are NOT fairly recent ones. Recent releases are eg "BlitzMax_rpi_arm64_0.146.3.58.202504270418.tar.xz"
- bcc is 0.146
- bmk is 3.58
- build date is 2025-04-27 (in the night, each weekly auto build is done sunday morning)

Fixes in modules (brl.mod) are not visible in the filename (brl.mod and the likes do not have a "module scoped" version ... you could just compare revisions). But often BCC fixes things or learns new stuff (new primitives like ulong etc) and modules get adapted to it (so from then on require the new BCC which only from then on supports the new primitive).

**Comment by meemoeuk2** [comment]
@GWRon 
thanks, I had a look at your link. I found some x86 releases in Dec 2024 and Jan 2025. I notice all of them are labelled 'pre-release'. I guess this is because the community is so small / informal that no one is beta-testing, so all releases are liable to be buggy.

**Comment by GWRon** [comment]
No..it is just that @woollybah did not make a new build (which he then would link on the frontpage).

There are a few users who run the latest builds (or almost latest) and thus indirectly test things. You only test what you are using... ;-)


But yes, userbase is quite small.

**Comment by meemoeuk2** [comment]
well, my project is halted for now cos the pre-packaged bundle found at  blitzmax compiler / parser is throwing a strange error at compile time from one its own files, not mine.
To make progress I need learn how to be a devops to transmit the latest version of blitz-ng bcc & bmk from github to my hard-disk and then get them functioning.
I've little skill/knowledge in this. I was able to download, install and run the pre-packed bundle from https://blitzmax.org/downloads/, but navigating the 100s of thousands of github files and spotting which files to download and where to put them on my hard-disk is largely a mystery. Any help would be appreciated. We can use this thread as a worklog of a noob to demo what its like trying to dl the latest version of BM-NG under the current github paradigm.

**Comment by GWRon** [comment]
Simply download one of the pre-releases for your operating system ..

https://github.com/bmx-ng/bmx-ng/releases

Unzip to where you want (the downloads are self-contained) and use it.

**Comment by meemoeuk2** [comment]
thanks, got version bcc 0.146 and bmk 3.58. That was easy.

During unzip windows kept saying 'don't have permission' and 'failed to copy' for .symlink files. They were all 0 byte files so I clicked 'ignore all'. The install still worked.

Other thing: 1.78GB. why so big? This is a lot more than blitzbasic on the Amiga that came on one 880K 3.5" disk.

The strange compile error i was getting with the older version of BM-NG is still here so I'll create a new issue thread. I've actually solved it now, however the 'internal' error message is likely undesirable, so brucey or hurryfish might want to fix it.

**Comment by GWRon** [comment]
I guess the Windows OS you are using did not come on 1.44 floppy discs ?

A usual cheapo SSD is at least 256gig, most will have > 1TB.
So we talk about <% of your storage. Given you had a 20 megs disc that time (on my 286 one it was 20) then I guess ratio is quite the same.


This aside: the majority of the 1.7gb is the C compiler. If you had that installed you could have trimmed down it quite a lot. Next is all the c libraries (sdl and the likes) which also add quite a bit.

Also these libs come with a lot of example source files (maybe even with media) and documentation.
I guess that BlitzBasic times came with a printed book... stuff is now commented in code etc.


A lot of this could be trimmed down for a way smaller package..but most of us would not bother to assist in such stuff (time vs effort).
